ArtOfIllusion/src/artofillusion/image/ExternalImage.java

```diff
@@ -0,0 +1,479 @@
+/* Copyright (C) 2017 by Petri Ihalainen
+
+   This program is free software; you can redistribute it and/or modify it under the
+   terms of the GNU General Public License as published by the Free Software
+   Foundation; either version 2 of the License, or (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful, but WITHOUT ANY 
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A 
+   PARTICULAR PURPOSE.  See the GNU General Public License for more details. */
+
+package artofillusion.image;
+
+import artofillusion.*;
+import artofillusion.math.*;
+import buoy.widget.BStandardDialog;
+import java.awt.*;
+import java.awt.image.*;
+import java.io.*;
+import java.util.Date;
+import javax.imageio.*;
+
+public class ExternalImage extends ImageMap
+{
+  private File externalFile;
+  private String loadingError, imageType;
+  private ImageMap imageMap;
```

Comment:
I don't see a problem with the current implementation.  ExternalImage is an ImageMap that delegates many of its functions to another ImageMap.  That seems simple enough.  It's performing a specialized function: creating a link between two different files doesn't belong in the parent ImageMap class.  But the image loaded from the external scene could be of many types, so it also needs a reference to that other object.  This is exactly the same way ExternalObject works.